### PR TITLE
make iframes link to rex

### PR DESCRIPTION
**Pull request by kswanson33**
openstax/cnx-recipes#4233

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+* Remove iframe baking from `BakeAutotitledNotes`
+* Define rex_link on `ElementBase`
+* Link to rex from iframe in `BakeIframes`
```
Comment (LostMojo) [inline CHANGELOG.md:11]: Are we not adding minor/major/patch to the end of changes now?

Comment (LostMojo) [reply CHANGELOG.md:11]: Cool! Thank you for clarifying

File: lib/kitchen/element_base.rb [hunk 2]
```diff
@@ -799,6 +799,31 @@ def as_enumerator
...
+
+      book_slug = document.search('span[data-type="slug"]').first[:'data-value']
+      chapter_count = element_with_ancestors.ancestor(:chapter).count_in(:book)
+      page_string = ''
+      page_title = ''
+      page = element_with_ancestors.ancestor(:page) if element_with_ancestors.has_ancestor?(:page)
+      if page && page&.is_introduction?
+        page_title = page.first('[data-type="document-title"]').text.strip.downcase
+      elsif page
+        page_string = "#{page.count_in(:chapter) - 1}-"
```
Comment (ajluk) [inline lib/kitchen/element_base.rb:817]: Can I ask why in the section page case such string was used - `page_in_chapter -1`. I was wondering just out of curiosity why it is always 1and not the numbering we use for sections, like `chapter_number.page_in_chapter_number`.

File: spec/kitchen_spec/directions/bake_iframes/v1_spec.rb [hunk 3]
```diff
@@ -3,76 +3,64 @@
...
           </div>
-        HTML
-      )
+        </div></div>
+      HTML
     )
   end
 
-  let(:book_with_note) do
+  let(:book_with_error_prone_iframe) do
```
Comment (ajluk) [inline spec/kitchen_spec/directions/bake_iframes/v1_spec.rb:50]: I'm not fully understand why in this book case the iframe is proning error :sweat_smile:

**Review by ajluk — CHANGES_REQUESTED** [review]
@kswanson33 I looked through it all and was concerned about only one place (composite pages title link text) and beside left some comments where I wasn't sure what's going on.

File: lib/kitchen/element_base.rb [hunk 4]
```diff
@@ -799,6 +799,31 @@ def as_enumerator
...
+
+      element_with_ancestors = document.book.chapters.search_with(
+        Kitchen::PageElementEnumerator, Kitchen::CompositePageElementEnumerator
+      ).search("##{id}").first
+
+      book_slug = document.search('span[data-type="slug"]').first[:'data-value']
+      chapter_count = element_with_ancestors.ancestor(:chapter).count_in(:book)
+      page_string = ''
+      page_title = ''
+      page = element_with_ancestors.ancestor(:page) if element_with_ancestors.has_ancestor?(:page)
```
[the remark this answers is not in the document]
Comment (kswanson33) [reply lib/kitchen/element_base.rb:813]: we're expecting either a page ancestor or a composite page ancestor to exist. this allows the composite page case to pass through without throwing an error.

File: spec/kitchen_spec/directions/bake_iframes/v1_spec.rb [hunk 5]
```diff
@@ -81,10 +64,54 @@
...
+    it 'warns when rex link can\'t be made - no slug' do
+      expect(Warning).to receive(:warn).with(
+        "Unable to find rex link for iframe with parent id=1234\n"
+      )
+      described_class.new.bake(book: book_with_iframe_no_slug)
+    end
+
+    it 'warns when rex link can\'t be made - no id' do
+      expect(Warning).to receive(:warn).with(
+        "Unable to find rex link for iframe with parent id=\n"
```
Comment (kswanson33) [inline spec/kitchen_spec/directions/bake_iframes/v1_spec.rb:112]: I know this error message isn't very helpful when the error is being caused by no IDs, but I'm not actually expecting this error case to come up very much

File: lib/kitchen/element_base.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (ajluk) [reply lib/kitchen/element_base.rb:817]: Thank you for clarifying! :relaxed:

**Review by ajluk — APPROVED** [review]
Looks really good, thank you for clarifying and adding more error checking test cases Kendra!

**Pull request merged**